### PR TITLE
GRW-888 / Feature / Pickup saved campaign code on offer page

### PR DESCRIPTION
**Pull request by robinandeer**
## What?

Replace manual cookie handling with `CampaignCode` interface.
Don't overwrite campaign code if already added to quote cart.

## Why?

We want to send links directly to the offer page (magic link) with a campaign code when retargeting users that never signed.

**Ticket(s): []**

### Before
Screenshot 2022-03-08 at 11 27 24

### After (reload)
Screenshot 2022-03-08 at 11 27 30

### CODE REVIEW

**Review by amandaflod — APPROVED** [review]
🪄

File: src/client/utils/campaignCode.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ const getPersistedCampaignCode = () => {
 }
 
 const saveCampaignCode = (campaignCode: string) => {
-  cookieStorage.setItem(CAMPAIGN_CODE_COOKIE_KEY, campaignCode)
+  cookieStorage.setItem(CAMPAIGN_CODE_COOKIE_KEY, campaignCode, { path: '/' })
```
Comment (guilhermespopolin) [inline src/client/utils/campaignCode.ts:12]: Does `{ path: '/' }` make the cookie availability through the whole app (every subdoman)? 🤔

File: src/client/pages/OfferNew/Introduction/Sidebar/DiscountCodeModal.tsx [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import styled from '@emotion/styled'
```
Comment (guilhermespopolin) [inline src/client/pages/OfferNew/Introduction/Sidebar/DiscountCodeModal.tsx:1]: There's a `CampaignCodeModal` under `Offer` folder. Should we change there as well?

File: src/client/utils/campaignCode.ts (hunk 1 above)
Comment (robinandeer) [reply src/client/utils/campaignCode.ts:12]: I don't think it applies to subdomains but the Cookie header will be sent on all URL paths:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#path_attribute

This is just how it was setup before, that's why I included it.

File: src/client/pages/OfferNew/Introduction/Sidebar/DiscountCodeModal.tsx (hunk 2 above)
Comment (robinandeer) [reply src/client/pages/OfferNew/Introduction/Sidebar/DiscountCodeModal.tsx:1]: It's already implemented there, no? In this function: `handleAddCampaignCode`

**Review by guilhermespopolin — APPROVED** [review]

**Pull request merged**